### ISSUE TITLE
Show Validation Errors for Quotes

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#3491]
**[Mingle Card: 3806](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3806)**
Narrative
---------

As a Sales Person, I can see validation errors on the quote.

Acceptance Criteria
-------------------

_All validations from the server should work on the mobile for editing quotes._

Analysis
--------

It is assumed for the purposes of this card, validation rules will be downloaded as JSON – from the value of the \`window.FX\_MobileRules.validationRules\` property defined by the  \`FX\_Mobile\_Rules\` static resource file (see #3909 for details) (see attached FX\_Mobile\_Rules file for example content of this file). 

In time, the rules from this source will used to supplement the Salesforce configured validation rules available from the tooling api (see card #3805), however, the rules from the tooling api are out-of-scope for now.

Given that the FX\_Mobile\_Rules file may define rules for any sobject type, we need to use this file from any form in mobile. The \`form.populateRaw()\` method from \`layout/form.js\` would seem like a reasonable place to ensure the validation rules defined in this file are available on the \`metadata\` object where they will be consumed.

The validation modules in Mobile (field-validator.js and form-validation) expect the validation rules to be available per field – rather than per object as is defined in the FX\_Mobile\_Rules file. Although the rules are available in their raw form on the \`metadata.validationRules\` object, it’s only the rules defined on each field (field.metadata.validationRules) that are actually applied by mobile. The complexity here is that in order to determine whether a validation rule applies to a field (either directly, or indirectly via a calculated field), the \`errorConditionFormula\` must be parsed (see [https://github.com/Liquidframeworks/alpine-fetch/blob/master/lib/sync/metadata.js#L11-L38).](https://github.com/Liquidframeworks/alpine-fetch/blob/master/lib/sync/metadata.js#L11-L38).​) Unless this parsing is performed by the the server, or performed once at sync time, the parsing of the validation rules will have to happen each time the form is loaded. We would need to evaluate the cost of parsing the rules at run-time to determine if it needs to be optimized/pre-computed. An alternative to making the validation rules available per-field, is to restructure/replace field-validator.js to work at the object level (rather than field level) using the raw FX\_Mobile\_Rules data – this might be preferable (for performance and simplicity).

» _An alternative to making the validation rules available per-field, is to restructure/replace field-validator.js to work at the object level (rather than field level) using the raw FX\_Mobile\_Rules data_

DP – I advocated for this when validation rules were first implemented.  Each validation rule could be a simple calculated value on the form.  When it’s value becomes true the object is not valid.  The fact that a field optionally can be assigned to a rule only tells us what extra UI behavior we need, not how validation should be bound.

DP – I suggest we add the new validation in parallel to avoid regression problems with the old method.  The old method will still evaluate but against an empty rules set.

Related Stories
---------------

The validation rules should be synced down based on card #3805 #3909.

Required Field Validation#3874

Tasks
-----

{{table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD}}

Defects
-------

{{ table query: SELECT Number, Name, Owner, 'Status' WHERE Type = Defect and 'Related Story' = THIS CARD }}

Test Plan
---------

**Comment by githubbob42** [comment]
**Pull Request #1711**

**Comment by githubbob42** [comment]
### Feature: #3593 Validations on Quotes ([Mingle](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/3911))
